cnxepub/models.py

```diff
@@ -128,12 +128,12 @@ def model_to_tree(model, title=None, lucent_id=TRANSLUCENT_BINDER_ID):
         {'id': <id>|'subcol', 'title': <title>, 'contents': [<tree>, ...]}
 
     """
-    if type(model) is TranslucentBinder:
+    id = model.ident_hash
+    if id is None and isinstance(model, TranslucentBinder):
```

Comment:
Hate to see a vision die, but to be honest, it's never been that clear. We waffled a lot about "zipped up, epub-like-not-really-epub-format" at the beginning. And cnx-epub does not in fact handle general pubs, following the spec: it doesn't create a spec compliant opf. It says in the README "Library for parsing and building EPUB3 files for connexions content." And we need a central library to enforce *Connexions* specific content limitations - i.e. presence of a license, at least one author, or the document having a unique ID. If cnx-epub isn't the place for that, where is? Not archive, I don't think - it's domain is persisting and serving the content. Ideally, it should know as little as possible about the internal structure. Perhaps publish.
I think there's probably a need for both - a generic epub handler, and code that enforces the Connexions/OpenStax specialization of that. I have a feeling cnx-epub might grow(split?) into that, as we put more  requirements in place  - creating user-facing epubs, importing arbitrary (broken?) epubs, dealing with epub editors. For the future, I'll try to see the overall shape of the code a bit better, see if we can nudge it that way.

P.S. this change (adding ids and metadata to a level of the data structure that did not have it before) is not minor surgery. I'm actually well pleased with how _little_ code changes are needed, mostly bypassing exceptions that had been coded in for the `subcol` special case. That speaks to the flexibility of the original design.